src/tactic/sanity_check.lean

```diff
@@ -152,70 +133,128 @@ do
     else if (d.is_definition : bool) then "is a def, should be a lemma/theorem"
     else "is a lemma/theorem, should be a def"
 
-/-- Print all declarations in mathlib incorrectly marked as def/lemma -/
-meta def incorrect_def_lemma_mathlib : tactic unit :=
-print_decls_mathlib unused_arguments >>= trace
-
 /-- Checks whether a declaration has a namespace twice consecutively in its name -/
 meta def dup_namespace (d : declaration) : tactic (option string) :=
 return $ let nm := d.to_name.components in if nm.chain' (≠) then none
   else let s := (nm.find $ λ n, nm.count n ≥ 2).iget.to_string in
   some $ "The namespace `" ++ s ++ "` is duplicated in the name"
 
+/-- Checks whether a declaration has a classical proof in the statement -/
```

Comment:
```suggestion
/-- Checks whether a declaration has a classical proof, or a `>` or `≥`, in the statement -/
```

Reply:
Actually, looking at this, it’s only the commented out code that checks for classical proofs, and the actual algorithm only looks for gt and ge. What’s going on here?

Reply:
First that declaration only checked for classicality, then it did both. Then I realized I needed a bunch more checks to avoid too many false positives for the classicality (for example, using `classical.prop_decidable` to show that `real` has `decidable_eq` is fine), so I removed that functionality for now (this commented out code will be temporary).

This command will be a lot easier to check with the modular `#sanity_check` I implemented in this PR, because then I can run in on all of mathlib, and tweak it multiple times, without recompiling mathlib each time.